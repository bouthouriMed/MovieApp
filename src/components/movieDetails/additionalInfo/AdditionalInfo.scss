@import "@/styles/variables";

.additional-info {
  margin-top: 2rem;
  padding: 2rem;
  background: rgba(18, 18, 18, 0.9);
  border-radius: 12px;
  position: relative;
  z-index: 1;

  h2 {
    margin-bottom: 1rem;
    color: $text-color;
  }

  .grid {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(220px, 1fr));
    gap: 1rem;

    div {
      font-size: 0.95rem;
      color: $text-color;

      strong {
        color: $primary-color;
      }
    }
  }

  .companies {
    margin-top: 2rem;

    h3 {
      margin-bottom: 1rem;
      color: $text-color;
    }

    .logos {
      display: flex;
      gap: 1rem;
      flex-wrap: wrap;

      .company {
        background: $secondary-color;
        padding: 0.5rem;
        border-radius: 8px;

        img {
          max-height: 40px;
          object-fit: contain;
        }

        span {
          font-size: 0.9rem;
          color: lighten($text-color, 30%);
        }
      }
    }
  }
}
