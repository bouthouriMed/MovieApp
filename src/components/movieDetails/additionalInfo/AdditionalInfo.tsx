import { memo } from "react";
import { Movie } from "src/pages/movieDetailPage/types";
import "./AdditionalInfo.scss";

interface AdditionalInfoProps {
  movie: Movie;
}

const AdditionalInfo = ({ movie }: AdditionalInfoProps) => {
  return (
    <div className="additional-info">
      <h2>Additional Information</h2>
      <div className="grid">
        <div>
          <strong>Spoken Languages:</strong>{" "}
          {movie?.spoken_languages?.map((l) => l?.english_name)?.join(", ")}
        </div>
        <div>
          <strong>Budget:</strong> ${movie?.budget?.toLocaleString()}
        </div>
        <div>
          <strong>Revenue:</strong> ${movie?.revenue?.toLocaleString()}
        </div>
        <div>
          <strong>Popularity:</strong> {movie?.popularity?.toFixed(0)}
        </div>
      </div>

      {movie.production_companies.length > 0 && (
        <div className="companies">
          <h3>Production Companies</h3>
          <div className="logos">
            {movie.production_companies.map((c) => (
              <div key={c.id} className="company">
                {c.logo_path ? (
                  <img
                    src={`https://image.tmdb.org/t/p/w200${c.logo_path}`}
                    alt={c.name}
                  />
                ) : (
                  <span>{c.name}</span>
                )}
              </div>
            ))}
          </div>
        </div>
      )}
    </div>
  );
};

export default memo(AdditionalInfo);
