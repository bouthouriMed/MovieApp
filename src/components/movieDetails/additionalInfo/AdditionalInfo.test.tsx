import { render } from "@testing-library/react";
import { mockedMovie } from "@/__mocks__/test_mocks";
import AdditionalInfo from "./AdditionalInfo";

describe("AdditionalInfo", () => {
  it("matches snapshot", () => {
    const { container } = render(<AdditionalInfo movie={mockedMovie} />);

    expect(container).toMatchSnapshot();
  });
});
