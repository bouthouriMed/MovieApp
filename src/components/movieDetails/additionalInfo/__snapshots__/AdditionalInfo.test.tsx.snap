// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdditionalInfo > matches snapshot 1`] = `
<div>
  <div
    class="additional-info"
  >
    <h2>
      Additional Information
    </h2>
    <div
      class="grid"
    >
      <div>
        <strong>
          Spoken Languages:
        </strong>
         
        English
      </div>
      <div>
        <strong>
          Budget:
        </strong>
         $
        1 000 000
      </div>
      <div>
        <strong>
          Revenue:
        </strong>
         $
        5 000 000
      </div>
      <div>
        <strong>
          Popularity:
        </strong>
         
        50
      </div>
    </div>
  </div>
</div>
`;
